ompi/communicator/communicator.h

```diff
@@ -125,10 +136,103 @@ OMPI_DECLSPEC OBJ_CLASS_DECLARATION(ompi_communicator_t);
 #define OMPI_COMM_BLOCK_OTHERS     8
 
 /* A macro comparing two CIDs */
-#define OMPI_COMM_CID_IS_LOWER(comm1,comm2) ( ((comm1)->c_contextid < (comm2)->c_contextid)? 1:0)
+#define OMPI_COMM_CID_IS_LOWER(comm1,comm2) ( ((comm1)->c_index < (comm2)->c_index)? 1:0)
 
-OMPI_DECLSPEC extern opal_pointer_array_t ompi_mpi_communicators;
+OMPI_DECLSPEC extern opal_hash_table_t ompi_comm_hash;
+OMPI_DECLSPEC extern opal_pointer_array_t ompi_comm_array;
```

Comment:
Why the name change ? The old name was clear enough.

Reply:
@hjelmn any reason not to go back to ompi_mpi_communicators?

Reply:
I can't remember why I changed the name. There probably was a reason at one point during the evolution of this branch. Most likely I was making the name consistent with ```ompi_comm_hash``` to make it clear that one is an array of communicators where the lookup is by index where the other one is a hash where the lookup is by exCID.

Reply:
If we remove this name change, the PR is getting more self-contained (aka. smaller and easier to understand).

Reply:
SGTM

Reply:
switched back to the old variable name